AK/Utf8View.cpp

```diff
@@ -224,14 +243,27 @@ u32 Utf8CodepointIterator::operator*() const
     size_t code_point_length_in_bytes = 0;
 
     bool first_byte_makes_sense = decode_first_byte(m_ptr[0], code_point_length_in_bytes, code_point_value_so_far);
-    if (!first_byte_makes_sense)
+
+    if (!first_byte_makes_sense) {
+        // The first byte of the codepoint doesn't make sense: output a replacement character
         dbgln("First byte doesn't make sense, bytes: {}", StringView { (const char*)m_ptr, m_length });
+        return 0xFFFD;
+    }
     VERIFY(first_byte_makes_sense);
-    if (code_point_length_in_bytes > m_length)
+
+    if (code_point_length_in_bytes > m_length) {
+        // There is not enough data left for the full codepoint: output a replacement character
         dbgln("Not enough bytes (need {}, have {}), first byte is: {:#02x}, '{}'", code_point_length_in_bytes, m_length, m_ptr[0], (const char*)m_ptr);
+        return 0xFFFD;
+    }
     VERIFY(code_point_length_in_bytes <= m_length);
 
     for (size_t offset = 1; offset < code_point_length_in_bytes; offset++) {
+        if (m_ptr[offset] >> 6 != 2) {
+            // One of the extension bytes of the codepoint doesn't make sense: output a replacement character
+            dbgln("Extension byte {:#02x} in {} position after first byte {:#02x} doesn't make sense.", m_ptr[offset], offset, m_ptr[0]);
+            return 0xFFFD;
+        }
         VERIFY(m_ptr[offset] >> 6 == 2);
```

Comment:
Same here.

Reply:
Same as above : this VERIFY is valid, even if trivial. The compiler will optimize it away, but it will defend against future canges to the code.